### ISSUE TITLE
Lint/NestedMethodDefinition shouldn't flag method definition in Rails #extending method

### ISSUE THREAD

**Issue by inkstak** [rubocop/rubocop#11018]
Rails is providing `extending` method to add methods to an ActiveRecord relation : 

https://api.rubyonrails.org/classes/ActiveRecord/QueryMethods.html#method-i-extending

```ruby
scope = Model.all.extending do
  def page(number)
    # pagination code goes here
  end
end
```

It'll be flagged with the following offense:

```
W: Lint/NestedMethodDefinition: Method definitions must not be nested. Use lambda instead.
  def page(number) ...
  ^^^^^^^^^^^^^^^^^
```

## Expected behavior

The cop should ignore `extending` method block.
If this is considered too specific to Rails, the cop should provide a configuration option to ignore given methods.

## RuboCop version
```
$ [bundle exec] rubocop -V
1.35.1 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.1.2 arm64-darwin21)
  - rubocop-performance 1.14.3
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.13.1
```

**Comment by inkstak** [comment]
Similar issue : [Lint/NestedMethodDefinition does not know about ActiveRecord association blocks](https://github.com/rubocop/rubocop/issues/8860)
